src/planet-a/SubdivTriangleMesh.cpp

```diff
@@ -0,0 +1,170 @@
+/**
+ * Open Space Program
+ * Copyright © 2019-2021 Open Space Program Project
+ *
+ * MIT License
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+#include "SubdivTriangleMesh.h"
+
+#include <Magnum/Math/Vector3.h>
+
+using namespace planeta;
+
+
+
+ChunkedTriangleMeshInfo planeta::make_subdivtrimesh_general(
+        unsigned int chunkMax, unsigned int subdivLevels, int pow2scale)
+{
+    // calculate stuff here
+
+    using std::sqrt;
+    using std::ceil;
+    using std::pow;
+
+    float const C = chunkMax;
```

Comment:
Why preconvert these to float? Using them as int should give you the behavior you want just as well, without forcing the compiler to pre-convert if there happens to be a form of the operations where float and int are faster?

Reply:
1. To shorten the name to improve readability of the equation
2. Slight concern over there being an integer implementation of std::sqrt